### ISSUE TITLE
Too many requests cause exception, while retry request should be done

### ISSUE THREAD

**Issue by maciejtreder** [serverless/serverless#4957]
# This is a Bug Report

## Description

For bug reports:
* After updating to version 1.27.1 I runned into 'Too many requests' error ([when using serverless-apigw-binary plugin](https://github.com/maciejtreder/serverless-apigw-binary))
* As in version 1.26.x - request should be retried
* Used config:
```
service: my-service

plugins:
  - serverless-apigw-binary

provider:
  name: aws
  runtime: nodejs6.10
  memorySize: 192
  timeout: 10
  stage: production
  region: eu-central-1

package:
  exclude:
   - src/**
   - node_modules/**
   - firebug-lite/**
   - e2e/**
   - coverage/**
   - '!node_modules/aws-serverless-express/**'
   - '!node_modules/binary-case/**'
   - '!node_modules/type-is/**'
   - '!node_modules/media-typer/**'
   - '!node_modules/mime-types/**'
   - '!node_modules/mime-db/**'

custom:
  apigwBinary:
    types:
      - '*/*'

functions:
  api:
    handler: lambda.universal
    events:
      - http: ANY {proxy+}
      - http: ANY /

```

* What stacktrace or error message from your provider did you see?
```
  Serverless Error ---------------------------------------
 
  Too Many Requests
 
  Stack Trace --------------------------------------------
 
ServerlessError: Too Many Requests
    at BbPromise.fromCallback.catch.err (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:262:11)
From previous event:
    at persistentRequest (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:250:13)
    at doCall (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:206:9)
    at BbPromise (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:219:14)
From previous event:
    at persistentRequest (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:204:38)
    at Object.request.requestQueue.add [as promiseGenerator] (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:240:49)
    at Queue._dequeue (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/promise-queue/lib/index.js:153:30)
    at /Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/promise-queue/lib/index.js:109:18
From previous event:
    at Queue.add (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/promise-queue/lib/index.js:94:16)
    at AwsProvider.request (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless/lib/plugins/aws/provider/awsProvider.js:240:39)
    at BinarySupport.createDeployment (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless-apigw-binary/src/index.js:30:26)
    at putSwagger.then (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless-apigw-binary/src/index.js:64:21)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
From previous event:
    at getApiId.then.apiId (/Users/mtreder/Documents/private/work/angular-universal-serverless/application/node_modules/serverless-apigw-binary/src/index.js:62:51)
    at <anonymous>
```

In version 1.26.1 output for same configuration is:

```
Serverless: Packaging service...
Serverless: Excluding development dependencies...
Serverless: Uploading CloudFormation file to S3...
Serverless: Uploading artifacts...
Serverless: Uploading service .zip file to S3 (1.29 MB)...
Serverless: Validating template...
Serverless: Updating Stack...
Serverless: Checking Stack update progress...
....................
Serverless: Stack update finished...
Service Information
service: binary-express-maciej-treder
stage: production
region: eu-central-1
stack: binary-express-maciej-treder-production
api keys:
  None
endpoints:
  ANY - https://k904suh2ag.execute-api.eu-central-1.amazonaws.com/production/{proxy+}
  ANY - https://k904suh2ag.execute-api.eu-central-1.amazonaws.com/production
functions:
  api: binary-express-maciej-treder-production-api
Serverless: 'Too many requests' received, sleeping 5 seconds
Serverless: 'Too many requests' received, sleeping 5 seconds
```

And application is deployed succesfully

**Comment by shavo007** [comment]
i get the same issue now 

serverless 1.27.2

**Comment by colinyip** [comment]
We've run into the same error with [serverless-prune-plugin](https://github.com/claygregory/serverless-prune-plugin) after upgrading to 1.27. 

This seems to be related the change made in this [pull request ](https://github.com/serverless/serverless/pull/4877). Looking at one of the error responses from AWS, we are getting status code 429, but the retryable flag is false. After the change, serverless won't retry for this kind of requests anymore.

This 429 error happens occasionally when we use the [serverless-prune-plugin](https://github.com/claygregory/serverless-prune-plugin). Can't seem to find a workaround fix at the moment, we are downgrading it back to 1.26 for now.

**Comment by HyperBrain** [comment]
I think the fix would just be to use `if (e.providerError && e.providerError.retryable || e.statusCode === 429) {` here https://github.com/medikoo/serverless/blob/77775f778f2eb598602f8559806ae8d0ed2f3fc1/lib/plugins/aws/provider/awsProvider.js#L209

**Comment by HyperBrain** [comment]
@colinyip @maciejtreder @shavo007 Can you try if the attached PR fixes the issue? Just use `"serverless": "github:serverless/serverless#4957-handle-429-errors"` in your package.json.

**Comment by colinyip** [comment]
@HyperBrain Also have tried the fix with "github:serverless/serverless#4957-handle-429-errors", seem to work fine 👍

**Comment by acambas** [comment]
I'm having this issue with serverless-plugin-custom-domain on the 1.27.2 version and it works fine on 1.26.1
